dashboard/app/models/script.rb

```diff
@@ -823,15 +827,15 @@ def get_script_level_by_id(script_level_id)
     script_levels.find(id: script_level_id.to_i)
   end
 
-  def get_script_level_by_relative_position_and_puzzle_position(relative_position, puzzle_position, lockable)
+  def get_script_level_by_relative_position_and_puzzle_position(relative_position, puzzle_position, use_survey_url)
     relative_position ||= 1
     script_levels.find do |sl|
       # make sure we are checking the native properties of the script level
       # first, so we only have to load lesson if it's actually necessary.
       sl.position == puzzle_position.to_i &&
         !sl.bonus &&
         sl.lesson.relative_position == relative_position.to_i &&
-        sl.lesson.lockable? == lockable
+        (use_survey_url ? !sl.lesson.numbered_lesson? : sl.lesson.numbered_lesson?)
```

Comment:
thank you, this is much better! however, I think there is still room for improvement:
1. this looks like it can be further simplified to `use_survey_url == !sl.lesson.numbered_lesson?`
2. even after addressing (1), it is still a bit more complicated than it needs to be to have two concepts which are opposites, roughly "survey" and "numbered". would you be willing to pick one of these and stick to it, so that you can go back to having something which looks more like the original code? then this line could read: `is_numbered_lesson == sl.lesson.numbered_lesson?` or `use_survey_url == sl.lesson.use_survey_url?`
3. it would be helpful to standardize on "survey" or "numbered" throughout the code you are touching in this PR.

Reply:
Thanks Dave for the push to make this change. Its definitely the right thing to update this. I have taken a pass out this. I used "unnumbered" and "numbered" lesson but let me know if you think there is a better wording to use